### PR TITLE
ocs: full deploy with customizations (fixes issue 7)

### PR DESCRIPTION
**Pull request by karmab**
fixes https://github.com/openshift-kni/install-scripts/issues/7
- deploy ocs using latest from master which defaults to csi and removes the need to handle secrets manually
- extend the use of envsubst to all the objects that might need customization
- optionally inject override object if rook_override env variable is set to true

### CODE REVIEW

**Comment by russellb** [comment]
Can you add the issue number to your commit message and PR summary?  #7

File: OCS/deploy-ocs.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+export namespace="openshift-storage"
+export rook_override="${rook_override:-false}"
+export public_network="${public_network:-1.1.1.1/24}"
+export cluster_network="${cluster_network:-1.1.1.1/24}"
+
+export rook_cluster=${namespace}
+
+oc patch featureGate cluster --type merge -p '{"spec":{"featureSet":"TechPreviewNoUpgrade"}}'
```
Comment (markmc) [inline OCS/deploy-ocs.sh]: Why is this required?

File: OCS/deploy-ocs.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+if [ ! -d rook ]; then
+    git clone https://github.com/rook/rook
+fi
+
+pushd .
+cd rook/cluster/examples/kubernetes/ceph
```
Comment (markmc) [inline OCS/deploy-ocs.sh]: Please include these manifests directly in this repository rather than pulling them from an upstream repository

Comment (markmc) [reply OCS/deploy-ocs.sh]: Actually no ... won't these manifests be in the OCS Operator?

File: OCS/deploy-ocs.sh [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+oc create -f operator-openshift-modified.yaml
+sleep 20
+
+oc wait --for condition=ready pod -l app=rook-ceph-operator -n ${namespace} --timeout=2400s
+sleep 20
+oc wait --for condition=ready pod -l app=rook-discover -n ${namespace} --timeout=2400s
+
+if [ "$rook_override" == "true" ] ; then
+  envsubst < ~1/rook-config-override.yaml | oc create -f -
```
Comment (markmc) [inline OCS/deploy-ocs.sh]: Not a fan of obscurities like `~1` -  a simple variable with the directory would be far more clear

**Comment by markmc** [comment]
I think we should start with an OCS operator install as per https://github.com/openshift/ocs-operator#install

File: OCS/deploy-ocs.sh (hunk 1 above)
Comment (karmab) [reply OCS/deploy-ocs.sh]: removed

File: OCS/deploy-ocs.sh (hunk 2 above)
Comment (karmab) [reply OCS/deploy-ocs.sh]: part of the operator indeed

File: OCS/deploy-ocs.sh (hunk 3 above)
Comment (karmab) [reply OCS/deploy-ocs.sh]: removed

File: OCS/deploy-ocs.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+while ! oc wait --for condition=ready pod -l name=ocs-operator -n ${NAMESPACE} --timeout=2400s; do sleep 10 ; done
+while ! oc wait --for condition=ready pod -l app=rook-ceph-operator -n ${NAMESPACE} --timeout=2400s; do sleep 10 ; done
+while ! oc wait --for condition=ready pod -l app=noobaa -n ${NAMESPACE} --timeout=2400s; do sleep 10 ; done
+
+if [ "$rook_override" == "true" ] ; then
+  envsubst < rook-config-override.yaml | oc create -f -
+fi
+
+oc create -f cluster.yaml
```
Comment (markmc) [inline OCS/deploy-ocs.sh]: Isn't the next step to create an OCS StorageCluster rather than a Rook CephCluster? As per https://github.com/openshift/ocs-operator#install

i.e. the OCS operator is responsible for creating the Ceph Cluster resource ...

Comment (karmab) [reply OCS/deploy-ocs.sh]: indeed but it currently misses elements needed for deployment on baremetal (local storage)

**Comment by karmab** [comment]
> Karim - can I assume this was tested the when install-scritps was working?

its what we ve been using so far

**Comment by jtaleric** [comment]
@karmab in your setups, this deploys ceph, but if you `oc rsh <toolbox>` and run `ceph -s`, is ceph in HEALTH_WARN due to number of PGs? 

This could be easily fixed to just set the pg count for the openshift-pool to 1024. Or do we want to enable the auto-scaler for pg count?

File: OCS/deploy-ocs.sh [hunk 5]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+NAMESPACE="openshift-storage"
+
+# export is required for envsubst
+export rook_override="${rook_override:-false}"
```
[the remark this answers is not in the document]
Comment (karmab) [reply OCS/deploy-ocs.sh]: because most of the folks consuming those assets dont have an appliance. furthermore, it should end up beeing done by the ocs operator

**Comment by karmab** [comment]
> @karmab in your setups, this deploys ceph, but if you `oc rsh <toolbox>` and run `ceph -s`, is ceph in HEALTH_WARN due to number of PGs?
> 
> This could be easily fixed to just set the pg count for the openshift-pool to 1024. Or do we want to enable the auto-scaler for pg count?

i have mix feelings on this one, i don't believe we really need to address those warnings just so it looks like nice on the ui.
worst case, ocs operator would do it?

**Comment by sreichar** [comment]
@jtaleric - okay to merge this this way for now

**Comment by mykaul** [comment]
> > @karmab in your setups, this deploys ceph, but if you `oc rsh <toolbox>` and run `ceph -s`, is ceph in HEALTH_WARN due to number of PGs?
> > This could be easily fixed to just set the pg count for the openshift-pool to 1024. Or do we want to enable the auto-scaler for pg count?
> 
> i have mix feelings on this one, i don't believe we really need to address those warnings just so it looks like nice on the ui.
> worst case, ocs operator would do it?

This is being taken care by the OCS team.

**Comment by jtaleric** [comment]
> > @karmab in your setups, this deploys ceph, but if you `oc rsh <toolbox>` and run `ceph -s`, is ceph in HEALTH_WARN due to number of PGs?
> > This could be easily fixed to just set the pg count for the openshift-pool to 1024. Or do we want to enable the auto-scaler for pg count?
> 
> i have mix feelings on this one, i don't believe we really need to address those warnings just so it looks like nice on the ui.
> worst case, ocs operator would do it?

Sorry this is a bit of misinformation, this is *NOT* just a UI issue. incorrect PG count is described here https://access.redhat.com/documentation/en-us/red_hat_ceph_storage/1.2.3/html/storage_strategies/pg-count 

I see @mykaul mentions OCS will handle it, so we should verify this.

**Comment by jtaleric** [comment]
> @jtaleric - okay to merge this this way for now

sure, I can open issues to track the PG count problem with the current implementation, so we don't lose that.

File: OCS/deploy-ocs.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (jtaleric) [reply OCS/deploy-ocs.sh]: @sreichar @karmab so are these scripts *FOR* the appliance or not? I keep hearing this flip on me. One day it is, another they are not? 

Additionally, how would the operator know to bind to a interface, the user has to provide that below -- highly suspect to think the OCS operator will just *know*.
